crates/wasi-provider/src/lib.rs

```diff
@@ -107,18 +113,48 @@ impl<S: ModuleStore + Send + Sync> Provider for WasiProvider<S> {
 
         let mut modules = self.store.fetch_pod_modules(&pod).await?;
         let client = kube::Client::from(self.kubeconfig.clone());
+        // TODO: Make volume refs part of the handle so they don't get dropped
+        let volumes = VolumeRef::volumes_from_pod(&self.volume_path, &pod, &client).await?;
         info!("Starting containers for pod {:?}", pod_name);
         for container in pod.containers() {
             let env = Self::env_vars(&container, &pod, &client).await;
             let module_data = modules
                 .remove(&container.name)
                 .expect("FATAL ERROR: module map not properly populated");
+            let container_volumes: HashMap<String, Option<String>> =
+                if let Some(volume_mounts) = container.volume_mounts.as_ref() {
+                    volume_mounts
+                        .iter()
+                        .map(|vm| -> anyhow::Result<(String, Option<String>)> {
+                            // Check the volume exists first
+                            let vol = volumes.get(&vm.name).ok_or_else(|| {
```

Comment:
Is it possible to move this closure to another function. This is slightly gnarly.

Reply:
Yeah, I can try to pop it out

Reply:
If we did this, we'd have to import `k8s_openapi` and add that direct dependency to our Cargo.toml, so I think I am ok without it. What do you think?